applications/luci-app-olsr-viz/htdocs/luci-static/resources/view/olsr-viz/olsr-viz-view.js

```diff
@@ -0,0 +1,173 @@
+					'.label {color:black;background-color:white}'
+				]));
+
+				var mainDiv = E('div', {
+					'id': 'main',
+					'style': 'width: 100%; height: 93%; border: 1px solid #ccc; margin-left:auto; margin-right:auto; text-align:center; overflow: scroll'
+				}, [
+					E('div', { 'id': 'edges', 'style': 'width: 1px; height: 1px; position: relative; z-index:2' }),
+					E('div', { 'id': 'nodes', 'style': 'width: 1px; height: 1px; position: relative; z-index:4' })
+				]);
+
+				var zoomInput = E('input', {
+					'id': 'zoom',
+					'name': 'zoom',
+					'type': 'text',
+					'value': '2.0',
+					'size': '5',
+					'style': 'min-width: unset !important;',
+					'onchange': 'set_scale()'
+				});
+				var metricInput = E('input', {
+					'id': 'maxmetric',
+					'name': 'maxmetric',
+					'type': 'text',
+					'value': '3',
+					'size': '4',
+					'style': 'min-width: unset !important;',
+					'onchange': 'set_maxmetric(this.value)'
+				});
+				var autoOptimizationCheckbox = E('input', {
+					'id': 'auto_declump',
+					'name': 'auto_declump',
+					'type': 'checkbox',
+					'onchange': 'set_autodeclump(this.checked)',
+					'checked': 'checked'
+				});
+				var hostnamesCheckbox = E('input', {
+					'id': 'show_hostnames',
+					'name': 'show_hostnames',
+					'type': 'checkbox',
+					'onchange': 'set_showdesc(this.checked)',
+					'checked': 'checked'
+				});
+
+				var form = E('form', { 'action': '' }, [
+					E('p', {}, [
+						E('b', { 'title': 'Bestimmt die Vergrößerungsstufe.' }, 'Zoom '),
+						E('a', { 'href': '#', 'onclick': 'set_scale(scale+0.1)' }, '+ '),
+						E('a', { 'href': '#', 'onclick': 'set_scale(scale-0.1)' },  '\u2212 '),
+						zoomInput,
+						E('b', { 'title': 'Beschränkt die Anzeige auf eine maximale Hop-Entfernung.' }, '&nbsp;&nbsp;Metrik'),
+						E('a', { 'href': '#', 'onclick': 'set_maxmetric(maxmetric+1)' }, '+ '),
+						E('a', { 'href': '#', 'onclick': 'if(0<maxmetric)set_maxmetric(maxmetric-1)' }, '\u2212'),
+						metricInput,
+						E('b', { 'title': 'Schaltet die automatischen Layout-Optimierung ein.' }, '&nbsp;&nbsp;Optimierung'),
+						autoOptimizationCheckbox,
+						E('b', { 'title': 'Zeige Hostnamen an.' }, ' |  Hostnamen'),
+						hostnamesCheckbox,
+						E('a', { 'href': '#', 'onclick': 'viz_save()', 'title': 'Speichert die aktuellen Einstellungen in einem Cookie.', 'style': 'font-weight:700;' }, '&nbsp;|&nbsp;&nbsp;Speichern'),
+						E('a', { 'href': '#', 'onclick': 'viz_reset()', 'title': 'Startet das Viz-Skriptprogramm neu.', 'style': 'font-weight:700;' }, '&nbsp;|&nbsp;&nbsp;Zur&uuml;cksetzen')
+					])
+				]);
+
+				var debugSpan = E('span', { 'id': 'debug', 'style': 'visibility:hidden;' });
+
+				var vizDiv = E('div', { 'id': 'RSIFrame', 'name': 'RSIFrame', 'style': 'border:0px; width:0px; height:0px; visibility:hidden;' });
+				var vizScript = E('script', { 'type': 'text/javascript' }, 'viz_setup("RSIFrame", "main","nodes", "edges");viz_update();');
```

Comment:
I suggest to modify `viz_setup()` to take the target div directly instead of a div name. Then simply replace this delayed script instantiation with a synchronous call to `viz_setup(vizDiv, "main", "nodes", "edges"); viz_update();`

and the `vizDiv` instantiation with `var vizDiv = E('div', { 'id': 'RSIFrame', 'name': 'RSIFrame', 'style': 'border:0px; width:0px; height:0px; visibility:hidden;' }, viz_res);

Reply:
I fixed the issues & also modified script file for directly using divs & inputs instead of getting them by id. and I implemented calling`viz_setup(vizDiv, "main", "nodes", "edges"); viz_update();` synchronously.

Reply:
And I will test it for some time more.

Reply:
I tested it & fixed all the issues. Thanks for the review.